### ISSUE TITLE
Shorter YAML arrays

### ISSUE THREAD

**Issue by jeffkreeftmeijer** [codebrawl/arnold#5]
When using `Arnold#edit` on a Mongoid model, the YAMLized version of the document looks like this:

```

---
_id: !ruby/object:BSON::ObjectId
  data:
  - 77
  - 255
  - 186
  - 158
  - 155
  - 230
  - 84
  - 12
  - 77
  - 0
  - 0
  - 1
email: jeff@kreeftmeijer.nl
github_id: 46321
login: jeffkreeftmeijer
name: Jeff Kreeftmeijer
nickname: jeffkreeftmeijer
token: 86b64b55edc4da8bb7f6b9c7272551402bfd13c1
urls:
  GitHub: http://github.com/jeffkreeftmeijer
  Blog: http://jeffkreeftmeijer.com
participations: []
average_score: 0.0
```

As you can see, the `_id` field in Mongoid is an array internally, which means YAML takes fifteen lines to show it. We've decided not to show all fields in #2, but since there will be an option to show these fields, we might need to build something in to handle this specific case. If I remember correctly, this would be valid YAML too:

```

---
_id: !ruby/object:BSON::ObjectId
  data: [77, 255, 186, 158, 155, 230, 84, 12, 77, 0, 0, 1]
email: jeff@kreeftmeijer.nl
github_id: 46321
login: jeffkreeftmeijer
name: Jeff Kreeftmeijer
nickname: jeffkreeftmeijer
token: 86b64b55edc4da8bb7f6b9c7272551402bfd13c1
urls:
  GitHub: http://github.com/jeffkreeftmeijer
  Blog: http://jeffkreeftmeijer.com
participations: []
average_score: 0.0
```

It would be awesome if we could get YAML to do that for any array field that doesn't contain arrays or hashes. :)

**Comment by razielgn** [comment]
I was looking into that yesterday: `to_yaml` takes an hash of options but none seem to work.
Apparently `data: [1, 2, 3, 4]` is called _inline_ format in yaml, but nothing seems to trigger that.

**Comment by jeffkreeftmeijer** [comment]
Yeah, we'll probably have to overwrite `to_yaml` a bit, since I only want to use the "inline" style for Arrays without other Arrays or Hashes. :)

**Comment by razielgn** [comment]
I couldn't even find where `to_yaml` is defined in yaml's code. It looks to be a nasty hack embedded somewhere.

**Comment by razielgn** [comment]
Oh nevermind, I've found the solution:

``` ruby
class Array
  def to_yaml_style; :inline; end
end
```

**Comment by jeffkreeftmeijer** [comment]
Yeah, found that a while back and tried to discuss it with you in #codebrawl but you were gone. ;)

`to_yaml_style` does exactly what we need. The only problem is that'd be monkey-patching `Array` and I really don't want to do that. We need to find a way to scope it so it only works for Arnold.

**Comment by jeffkreeftmeijer** [comment]
9727af92e0b5d442579758d5e74c9bcab1f20018 adds the `Array` patch, but modifies the global namespace. I'll keep this one open until we find a solution for this. Any ideas? :)

**Comment by jeffkreeftmeijer** [comment]
Great, I had to revert 6916156a25f481029c28e6d3e1b7e42dd2e3df72 and 9727af92e0b5d442579758d5e74c9bcab1f20018, since they broke the build on 1.9.2 and 1.9.3. It turns out `to_yaml_style` only works in 1.8.x. :\

**Comment by jeffkreeftmeijer** [comment]
Closed by dda8a55557b24612483ddd630a4595670a68d255. Since YAML on multiple Rubies is hell, this is quite a nasty patch. I started #6 for it to clean up a bit.

I'm glad to finally be able to close this. :)
